cmd/gateway-gcs.go

```diff
@@ -908,29 +910,40 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 	if err != nil {
 		return ObjectInfo{}, gcsToObjectError(traceError(err), bucket, key)
 	}
+
+	if len(uploadedParts) > maxPartCount {
+		return ObjectInfo{}, traceError(NotSupported{})
+	}
+
 	r, err := object.NewReader(l.ctx)
```

Comment:
We can use `defer r.Close()` instead of closing the `ReadCloser` in every failure branch individually.